docs/guide/extending/custom_fieldtype_comparison.md

```diff
@@ -0,0 +1,117 @@
+{
+    return new TextLine([
+        'textLine' => new StringComparisonValue([
+            'value' => $value->text,,
+        ]),
+    ]);
+}
+```
+
+Also, the Field Type requires additional configuration (e.g. `comparable_fieldtypes.yaml`).
+An exemplary configuration, in this case for `ezstring`, looks the following way:
+
+``` yaml
+EzSystems\EzPlatformContentComparison\FieldType\TextLine\Comparable:
+  tags:
+    - { name: ezplatform.field_type.comparable, alias: ezstring }
+```
+
+## Comparison engine
+
+The comparison engine handles the operations required for comparing the contents of Field Types.
+Note that each Field Type requires a separate comparison engine.
+
+When creating a custom engine, ensure that it implements the `EzSystems\EzPlatformContentComparison\Engine` interface:
+
+``` php
+declare(strict_types=1);
+
+namespace EzSystems\EzPlatformContentComparison\Engine;
+
+use EzSystems\EzPlatformContentComparison\Comparison\ComparisonData;
+```
+
+The engine requires also configuration similar to the following one:
+
+``` yaml
+EzSystems\EzPlatformContentComparison\Engine\Value\StringValueComparisonEngine: ~
+
+EzSystems\EzPlatformContentComparison\Engine\FieldType\TextLineComparisonEngine:
+  tags:
+    - { name: ezplatform.field_type.comparable.engine, supported_type: EzSystems\EzPlatformContentComparison\Comparison\Field\TextLine }
+
+EzSystems\EzPlatformContentComparison\Engine\NoComparisonValueEngine:
+  tags:
+    - { name: ezplatform.field_type.comparable.engine, supported_type: EzSystems\EzPlatformContentComparison\Comparison\Field\NoComparison }
+
+EzSystems\EzPlatformContentComparison\FieldType\NonComparable:
+  tags:
+    - { name: ezplatform.field_type.comparable, alias: eznoncomparable }
+```
+
+Lines 3-5 and 7-9 configure two [comparison engines](#comparison-engine) required for handling the content of the `ezstring` Field Type.
+When configuring the engines, ensure that you provided the `supported_type` tag.
+Lines 11-13 are for providing the configuration in case the Field Type versions are not comparable.
+Finally, in lines 15-17 you configure the Field Type to be comparable by using the `alias` tag.
+
+!!! note
+
+    There is a difference in tagging:
+
+    - Configure the engines with the `ezplatform.field_type.comparable.engine` tag
+    - Configure the Field Type with the `ezplatform.field_type.comparable` tag
+
+### VersionDiff
+
+For the comparison purposes, `VersionDiff` is built by `ContentComparisonService::CompareVersions`.
+It consists of array of `EzSystems\EzPlatformContentComparison\Values\FieldValueDiff`.
+It is an object that holds `EzSystems\EzPlatformContentComparison\Result\ComparisonResult`,
+which is specific to a Field Type, as different Field Types have distinct way of showing the difference between their versions. 
+
+The `VersionDiff` is used by the `EzSystems\EzPlatformContentComparison\Service\ContentComparisonService` containing the following method:
```

Comment:
this sentence is now not needed, one as it not true, and twice is is duplicated but first sentence in this paragraph.